doc/source/ray-air/computer-vision.rst

```diff
@@ -0,0 +1,285 @@
+
+    Create two ``TorchVisionPreprocessors`` -- one to normalize images, and another to
+    augment images. Later, you'll pass the preprocessors to ``Trainers``, ``Predictors``,
+    and ``PredictorDeployments``.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_preprocessors_start__
+        :end-before: __torch_preprocessors_stop__
+        :dedent:
+
+    .. warning::
+        If you read raw images in `Reading image data`_, then your dataset
+        contains class names instead of integer targets. To fix this issue, chain
+        ``preprocessor`` with a label encoder:
+
+        .. code-block:: python
+
+            from ray.data.preprocessors import Chain, LabelEncoder
+
+            preprocessor = Chain(
+                preprocessor,
+                LabelEncoder(columns=["label"])
+            )
+
+
+.. tabbed:: TensorFlow
+
+    .. testcode::
+
+        from tensorflow.keras.applications import imagenet_utils
+
+        from ray.data.preprocessors import BatchMapper
+
+        def preprocess(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+            batch["image"] = imagenet_utils.preprocess_input(batch["image"])
+            return batch
+
+        preprocessor = BatchMapper(preprocess, batch_format="numpy")
+
+
+For more information on transforming data, see Using Preprocessors and Transforming Datasets.
+
+Training vision models
+----------------------
+
+:class:`Trainers <ray.train.trainer.BaseTrainer>` let you train models in parallel.
+
+.. tabbed:: Torch
+
+    To train a vision model, define the training loop per worker.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_training_loop_start__
+        :end-before: __torch_training_loop_stop__
+        :dedent:
+
+    Then, create a :class:`~ray.train.torch.TorchTrainer` and call
+    :meth:`~ray.train.torch.TorchTrainer.fit`.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_trainer_start__
+        :end-before: __torch_trainer_stop__
+        :dedent:
+
+    For a more in-depth example, read :doc:`/ray-air/examples/torch_image_example`.
+
+.. tabbed:: TensorFlow
+
+    ham
+
+Creating checkpoints
+--------------------
+
+:class:`Checkpoints <ray.air.checkpoint.Checkpoint>` are required for batch inference and model
+serving. They contain model state and optionally a preprocessor.
+
+.. tabbed:: Torch
+
+    To create a :class:`~ray.train.torch.TorchCheckpoint`, pass a Torch model and
+    the :class:`~ray.data.preprocessor.Preprocessor` you created in `Transforming images`_
+    to :meth:`TorchCheckpoint.from_model() <ray.train.torch.TorchCheckpoint.from_model>`.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_checkpoint_start__
+        :end-before: __torch_checkpoint_stop__
+        :dedent:
+
+.. tabbed:: TensorFlow
+
+    ham
+
+.. tip::
```

Comment:
instead of having this as a tip, can we explain this in the top-level of the section?

As part of the same section as 

> :class:`Checkpoints <ray.air.checkpoint.Checkpoint>` are required for batch inference and model
serving. They contain model state and optionally a preprocessor.